libs/ml/tests/ml/ops/divide.cpp

```diff
@@ -0,0 +1,69 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "ml/ops/divide.hpp"
+
+#include "math/tensor.hpp"
+#include "vectorise/fixed_point/fixed_point.hpp"
+
+#include "gtest/gtest.h"
+
+#include <cstdint>
+#include <vector>
+
+template <typename T>
+class DivideTest : public ::testing::Test
+{
+};
+
+using MyTypes = ::testing::Types<fetch::math::Tensor<float>, fetch::math::Tensor<double>,
+                                 fetch::math::Tensor<fetch::fixed_point::FixedPoint<16, 16>>,
+                                 fetch::math::Tensor<fetch::fixed_point::FixedPoint<32, 32>>>;
+
+TYPED_TEST_CASE(DivideTest, MyTypes);
+
+TYPED_TEST(DivideTest, forward_test)
+{
+  TypeParam data = TypeParam::FromString("1, 1, 2; 3, 4, 5");
+
+  TypeParam gt = TypeParam::FromString("1, 1, 1; 1, 1, 1");
+
+  fetch::ml::ops::Divide<TypeParam> op;
+
+  TypeParam prediction(op.ComputeOutputShape({data}));
+  op.Forward({data, data}, prediction);
```

Comment:
this is not a great test. dividing values by themselves and checking the output is 1 does not really test that much. we should check:
1. do some reasonable values give function tolerance precision results.
2. what happens for division by 0?
3. negative values?